File: search2d.java
// Time Complexity : o(m+n)
// Space Complexity : o(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no




// Your code here along with comments explaining your approach


class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length;
        int n=matrix[0].length;
        int i=0,j=n-1;
        while(i<m && j>=0){
            if(matrix[i][j]>target){
                j--;
            }else if(matrix[i][j]<target){
                i++;
            }else{
                return true;
            }
            
        }
        return false;
    }
}

File: search2d.py
'''
Time Complexity: O(m+n)
Space Complexity: O(1)
'''
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        row = 0
        rowlen = len(matrix)
        collen = len(matrix[0])
        col = collen-1
        while(row>=0 and row<rowlen and col>=0 and col<collen):
            #print(matrix[row][col])
            if(matrix[row][col]==target):
                return True
            elif(matrix[row][col]>target):
                col -=1
            else:
                row += 1
        return False